src/plugins/markdown.ts

```diff
@@ -5,67 +5,156 @@
-                }
+                // using `reduce` so we can add one or many entries for each node
+                page.contents = page.contents.reduce((array, content) => {
+                    if (typeof content === "string" || content.tag !== "include") {
+                        return array.concat(content);
+                    }
+                    // inline @include page
+                    try {
+                        const pageToInclude = pageStore.remove(content.value);
+                        return array.concat(pageToInclude.contents);
+                    } catch (ex) {
+                        throw new Error(`Unknown @include reference '${content.value}' in '${page.reference}'`);
+                    }
+                }, [] as StringOrTag[]);
                 return page;
             });
-        const docs = mapToObject(pageStore);
-        return { docs };
+        return pageStore;
+    }
+
+    private buildNavTree(pages: PageMap) {
+        // navPage is used to construct the sidebar menu
+        const roots = createNavigableTree(pages, pages.get(this.options.navPage)).children as IPageNode[];
+        // nav page is not a real docs page so we can remove it from output
+        pages.remove(this.options.navPage);
+        roots.forEach((page) => {
+            if (isPageNode(page)) {
+                page.children.forEach((child) => this.nestChildPage(pages, child, page));
+            }
+        });
+
+        return roots;
+    }
+
+    private nestChildPage(pages: PageMap, child: IPageNode | IHeadingNode, parent: IPageNode) {
+        const originalRef = child.reference;
+
+        // update entry reference to include parent reference
+        const separator = isPageNode(child) ? this.options.pageSeparator : this.options.headingSeparator;
+        const nestedRef = [parent.reference, slugify(originalRef)].join(separator);
+        child.reference = nestedRef;
+
+        if (isPageNode(child)) {
+            // rename nested pages to be <parent>.<child> and remove old <child> entry
+            const page = pages.remove(originalRef);
+            page.reference = nestedRef;
+            pages.set(nestedRef, page);
+            // recurse through page children
+            child.children.forEach((innerchild) => this.nestChildPage(pages, innerchild, child));
+        }
     }
 }
 
-function mapToObject<T>(map: Map<string, T>): { [key: string]: T } {
-    const object: { [key: string]: T } = {};
-    for (const [key, val] of map.entries()) {
-        object[key] = val;
+function createNavigableTree(pages: PageMap, page: IPageData, depth = 0) {
+    const pageNode: IPageNode = initPageNode(page, depth);
+    if (page.contents != null) {
+        page.contents.forEach((node: StringOrTag, i: number) => {
+            if (isTag(node)) {
+                if (node.tag === "page") {
+                    const subpage = pages.get(node.value);
+                    // if (subpage === undefined) {
```

Comment:
it's a reminder! 
the error in `.get()` supersedes all my other errors which were really nice and contextual. wondering if `.get()` should return `IPageData | undefined` and force me to handle errors here?